lib/openxml/pptx/properties/slide_layout_id.rb

```diff
@@ -0,0 +1,26 @@
+module OpenXml
+  module Pptx
+    module Properties
+      class SlideLayoutId < OpenXml::Properties::ComplexProperty
+        include OpenXml::ContainsProperties
+        namespace :p
+        tag :sldLayoutId
+
+        attribute :id, expects: :integer, in_range: (2_147_483_648..4_294_967_295)
+        attribute :rid, displays_as: :id, namespace: :r, expects: :string, required: true
+
+        property :extension_list
+
+        def initialize(*_args)
+          super
+          generate_id
+        end
+
+        def generate_id
+          self.id = 2**31 + object_id % 2**31
```

Comment:
Should this be something like `OpenXml::Pptx::MAX_ID_SIZE`? It looks like a function of that...